environs/networking.go

```diff
@@ -42,6 +42,18 @@ type Networking interface {
 	// provider that have subnets available.
 	Spaces() ([]network.SpaceInfo, error)
 
+	// ProviderSpaceInfo returns the details of the space requested as
+	// a ProviderSpaceInfo. This will contain everything needed to
+	// decide whether an Environ of the same type in another
+	// controller could route to the space. If "" is passed as
+	// ProviderSpaceId, details for the default space will be
+	// returned.
+	ProviderSpaceInfo(providerSpaceId string) (ProviderSpaceInfo, error)
+
+	// IsRoutable returns whether this Environ can connect to the
+	// given space via cloud-local addresses.
+	IsRoutable(targetSpace ProviderSpaceInfo) (bool, error)
```

Comment:
Is this the second comment to one that never got added? I'm guessing it was "make `ProviderSpaceInfo` return a pointer"?

Will do on both counts.